tests/manage/z_cluster/test_multiple_mon_pods_stays_on_same_node.py

```diff
@@ -0,0 +1,215 @@
+
+                # Wait untill mon pod recovery
+                POD_OBJ.wait_for_resource(
+                    condition=STATUS_RUNNING,
+                    selector=MON_APP_LABEL,
+                    resource_count=len(mon_pod),
+                    timeout=3600,
+                    sleep=5,
+                )
+                log.info("All mons are up and running")
+
+                # Check the ceph health OK
+                ceph_health_check(tries=90, delay=15)
+
+        request.addfinalizer(finalizer)
+
+    def mon_pods_running_on_same_node(self):
+        """
+        Verifies two mons are running on same node
+
+        """
+        mon_running_nodes = get_mon_running_nodes()
+        dup = set()
+        mons_on_same_node = [m for m in mon_running_nodes if m in dup or dup.add(m)]
+        assert mons_on_same_node == [], "Two or more mons running on same node"
+        log.info("Mons are running on different nodes")
+
+    def test_multiple_mon_pod_stays_on_same_node(self):
+        """
+        A testcase to verify multiple mon pods stays on same node
+
+        1. Edit the rook-ceph-mon-endpoints configmap
+           say, assign mon-a to another node that would be on
+           the same node as another mon (compute-1 instead of compute-0)
+        2. Delete the mon-a deployment
+        3. Edit the mon-b deployment to remove the required mon anti-affinity
+        4. Restart the operator
+        5. Edit the mon-a deployment to remove the required mon anti-affinity
+        6. See mon-a start on compute-1 with mon-b
+        7. Soon after, see the operator failover one of these mons onto the
+        node that doesn't currently have a mon (compute-0) and start mon-d
+
+        """
+        # Initialize
+        rook_ceph_mon = "rook-ceph-mon"
+
+        # Get mons running on pod
+        mon_pods = get_mon_pods()
+        mon_name_to_del = mon_pods[0].get().get("metadata").get("labels").get("mon")
+        mon_name_to_edit = mon_pods[1].get().get("metadata").get("labels").get("mon")
+        mon_node = get_pod_node(mon_pods[1])
+
+        # Edit the rook-ceph-mon-endpoints
+        log.info(f"Edit the configmap {ROOK_CEPH_MON_ENDPOINTS}")
+        configmap_obj = OCP(kind=CONFIGMAP, namespace=OPENSHIFT_STORAGE_NAMESPACE)
+        rook_ceph_mon_configmap = configmap_obj.get(
+            resource_name=ROOK_CEPH_MON_ENDPOINTS
+        )
+        json_val = json.loads(rook_ceph_mon_configmap["data"]["mapping"])
+        json_val["node"][mon_name_to_del].update(json_val["node"][mon_name_to_edit])
+        rook_ceph_mon_configmap["data"]["mapping"] = json.dumps(json_val)
+        new_data = rook_ceph_mon_configmap["data"]
+        params = f'{{"data": {json.dumps(new_data)}}}'
+        configmap_obj.patch(
+            resource_name=ROOK_CEPH_MON_ENDPOINTS,
+            params=params,
+            format_type="strategic",
+        )
+        log.info(f"Configmap {ROOK_CEPH_MON_ENDPOINTS} edited successfully")
+
```

Comment:
Lets also dump the mon config maps yamls to the console so that we can be sure that the changes are successful and it can also help during debugging the test failure

Reply:
done